edx_toggles/toggles/testutils.py

```diff
@@ -68,15 +65,13 @@ def __init__(self, switch, active):
         self.switch = switch
         self._cached_value = None
         self._previous_active = None
-        super().__init__(switch.namespaced_switch_name, active)
+        super().__init__(switch.name, active)
 
     def __enter__(self):
-        self._previous_active = self.switch.waffle_namespace.is_enabled(
-            self.switch.switch_name
-        )
-        self.switch.waffle_namespace.set_request_cache(self.switch.namespaced_switch_name, self.active)
+        self._previous_active = self.switch.is_enabled()
```

Comment:
I didn't noticed this earlier because I wasn't looking at `override_waffle_flag` and `override_waffle_switch` code side-by-side, but why are these implemented differently?
* Is there an actual difference?
* If not, could we choose your favorite implementation and duplicate it?
* Can we get the docs more consistent and complete here?
This could be a follow-up if it isn't quick.

Reply:
> Is there an actual difference?

Yes. The `override_waffle_switch` makes an explicit call to `WaffleSwitch.is_enabled`, which fills the cache, while `override_waffle_flag` only looks at the cached values. At this point I'd rather avoid changing the behaviour of these methods, which are quite sensitive.

Reply:
Makes sense not to touch right now.  Maybe a (potentially duplicated) docstring?
```
Note: The implementation for overriding switches and flags differ. The ``override_waffle_switch``
  makes an explicit call to ``WaffleSwitch.is_enabled``, which fills the cache, while
  ``override_waffle_flag`` only looks at the cached values. It is unclear whether this difference
  is important, or just due to being developed at different times by different people.
```
Maybe I over explain, but when I have a question on a PR, I often find myself re-asking  the same question in the future and not remembering the answer. :)